source/shared/core_stmt.cpp

```diff
@@ -922,20 +920,18 @@ void core_sqlsrv_get_field( sqlsrv_stmt* stmt, SQLUSMALLINT field_index, sqlsrv_
 		// if the field is to be cached, and this field is being retrieved out of order, cache prior fields so they
 		// may also be retrieved.
 		if( cache_field && (field_index - stmt->last_field_index ) >= 2 ) {
-			sqlsrv_phptype invalid;
-			invalid.typeinfo.type = SQLSRV_PHPTYPE_INVALID;
-			for( int i = stmt->last_field_index + 1; i < field_index; ++i ) {
-				SQLSRV_ASSERT((cached = reinterpret_cast<field_cache*>(zend_hash_index_find_ptr(Z_ARRVAL(stmt->field_cache), i))) == NULL,
-					"Field already cached." );
-				core_sqlsrv_get_field( stmt, i, invalid, prefer_string, field_value, field_len, cache_field,
-									sqlsrv_php_type_out TSRMLS_CC );
-				// delete the value returned since we only want it cached, not the actual value
-				if( field_value ) {
-					efree( field_value );
-					field_value = NULL;
-					*field_len = 0;
-				}
-			}
+                   sqlsrv_phptype invalid;
+                   invalid.typeinfo.type = SQLSRV_PHPTYPE_INVALID;
+                   for( int i = stmt->last_field_index + 1; i < field_index; ++i ) {
+                       SQLSRV_ASSERT( reinterpret_cast<field_cache*>( zend_hash_index_find_ptr( Z_ARRVAL( stmt->field_cache ), i )) == NULL, "Field already cached." );
+                       core_sqlsrv_get_field( stmt, i, invalid, prefer_string, field_value, field_len, cache_field, sqlsrv_php_type_out TSRMLS_CC );
```

Comment:
In quite a lot of places 'NULL' is replaced by '0'. Why don't we do the same here? 
Also, I assume the variable 'cached' is not necessary?